assignment1.txt

```diff
@@ -0,0 +1,6 @@
+A compiler is basically a translator that transforms what the user codes through modern day coding languages into binary/machine code, or a quintessential language that the computer can understand.
```

Comment:
right, more specifically, a compiler takes your code written in C, removes the whitespace, places this into a parse tree, optimizes it, and then takes that parse tree and translates it into assembly code.